.build/build.ps1

```diff
@@ -0,0 +1,191 @@
+task UpdateVersionInfo GenerateVersionInformation, {
+    "Updating assembly information"
+
+    # Ignore anything under the Testing/ folder
+    @(Get-ChildItem "$RootDir" AssemblyInfo.cs -Recurse) | where { $_.FullName -notlike "$RootDir\Testing\*" } | ForEach {
+        Update-AssemblyVersion $_.FullName `
+            -Version $script:AssemblyVersion `
+            -FileVersion $script:AssemblyFileVersion `
+            -InformationalVersion $script:NuGetPackageVersion
+    }
+}
+
+# Synopsis: Update the nuspec dependencies versions based on the versions of the nuget packages that are being used
+task UpdateNuspecVersionInfo {
+    # Find all the packages.config
+    $projectFiles = Get-ChildItem "$RootDir" -Recurse -Filter "*.csproj" `
+                      | ?{ $_.fullname -notmatch "\\(.build)|(packages)\\" } `
+                      | Resolve-Path
+
+    # Update dependency verions in each of our .nuspec file based on what is in our packages.config
+    Resolve-Path "$RootDir\Nuspec\*.nuspec" | Update-NuspecDependenciesVersions -ProjectFilePaths $projectFiles -SpecificVersions -verbose
+}
+
+# Synopsis: A task that makes sure our initialization tasks have been run before we can do anything useful
+task Init CreateFolders, RestoreNugetPackages, GenerateVersionInformation
+
+# Synopsis: Compile the Visual Studio solution
+task Compile Init, UpdateVersionInfo, {
+    use 15.0 MSBuild
+    try {
+        exec {
+            & MSBuild `
+                "$Solution" `
+                /maxcpucount `
+                /nodereuse:false `
+                /target:Build `
+                /p:Configuration=$Configuration `
+                /flp1:verbosity=normal`;LogFile=$LogsDir\_msbuild.log.normal.txt `
+                /flp2:WarningsOnly`;LogFile=$LogsDir\_msbuild.log.warnings.txt `
+                /flp3:PerformanceSummary`;NoSummary`;verbosity=quiet`;LogFile=$LogsDir\_msbuild.log.performanceSummary.txt `
+                /flp4:verbosity=detailed`;LogFile=$LogsDir\_msbuild.log.detailed.txt `
+                /flp5:verbosity=diag`;LogFile=$LogsDir\_msbuild.log.diag.txt `
+        }
+    } finally {
+        TeamCity-PublishArtifact "$LogsDir\_msbuild.log.* => logs/msbuild.$Configuration.logs.zip"
+    }
+}
+
+# Synopsis: Execute our unit tests
+task UnitTests {
+    "$RootDir\bin\net461\*.Test.dll" | Resolve-Path | ForEach-Object {
+        try {
+            Invoke-NUnitForAssembly `
+                -AssemblyPath $_ `
+                -NUnitVersion "2.6.4" `
+                -FrameworkVersion "net-4.0" `
+                -EnableCodeCoverage $false `
+                -ExcludedCategories @('Explicit') `
+        }
+        catch {
+            Write-Host "Error while running tests: $_"
+        }
+    }
+}
+
+# Synopsis: Build the nuget packages.
+task BuildNugetPackages Init, UpdateNuspecVersionInfo, {
+    New-Item $NugetPackageOutputDir -ItemType Directory -Force | Out-Null
+
+    $escaped=$ReleaseNotes.Replace('"','\"')
```

Comment:
There's a cmdlet in RedGate.Build, `ConvertTo-ShellEscaped`, that does this following the correct rules for shell escaping in Windows.

Reply:
I copied this from the build script template repo. I'll try it and see if it works.